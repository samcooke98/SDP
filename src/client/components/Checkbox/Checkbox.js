import React from "react";
import styles from "./web.css";

export default class Checkbox extends React.Component {
    render() {
        return (
            <label
                htmlFor={this.id}
                className={styles.label}
            >
                <input
                    id={this.id}
                    type="checkbox"
                    className={styles.checkbox}
                />
                {this.props.title}
            </label>
        )

    }
}