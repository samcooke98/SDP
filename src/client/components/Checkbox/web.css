input[type=checkbox] {
	display:inline;
	width: 13px;
	height: 13px;
	padding: 0;
	margin:0;
	margin-right: 10px;
	vertical-align: bottom;
	position: relative;
	top: -2px;
	*overflow: hidden;
}

.label {
	font-family: 'Raleway';
	display: block;
	padding-left: 40px;
  	text-indent: -15px;
  	margin: 5px;
}