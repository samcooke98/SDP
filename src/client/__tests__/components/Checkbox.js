import React from "react";
import renderer from "react-test-renderer";
// import ReactDOM from 'react-dom';

import Checkbox from "../../components/Checkbox/Checkbox";


describe('React Component: Checkbox', () => {
    it("matches a snapshot", () => {
        const tree = renderer.create(
            <Checkbox />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})