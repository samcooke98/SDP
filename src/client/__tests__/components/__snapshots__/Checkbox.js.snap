// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component: Checkbox matches a snapshot 1`] = `
<label
  className={undefined}
  htmlFor={undefined}
>
  <input
    className={undefined}
    id={undefined}
    type="checkbox"
  />
</label>
`;
